stores.js

```diff
@@ -0,0 +1,179 @@
+    hash(key, value)
+    // set locally for easy cached retrieval
+    store.set(key.toString('hex'), value)
+    // send to the dht
+    this.update('immutable-store', key, value, (err) => {
+      if (err) {
+        cb(err)
+        return
+      }
+      cb(null, key)
+    })
+  },
+  command: {
+    update ({ target, value }, cb) {
+      if (value == null) {
+        cb(null)
+        return
+      }
+      const key = target.toString('hex')
+      store.set(key, value)
+      cb(null)
+    },
+    query ({ target }, cb) {
+      cb(null, store.get(target.toString('hex')))
+    }
+  }
+})
+
+const mutable = (store) => ({
+  keypair () {
+    const pk = Buffer.alloc(pkSize)
+    const sk = Buffer.alloc(skSize)
+    createKeypair(pk, sk)
+    return { pk, sk }
+  },
+  get (key, opts, cb = opts) {
+    if (typeof cb !== 'function') throw Error('Callback is required')
+    if (typeof opts !== 'object') throw Error('Options are required')
+    const { salt, seq } = opts
+    if (typeof seq !== 'number') throw Error('seq is a required option')
+    if (salt && !Buffer.isBuffer(key)) throw Error('salt must be a buffer')
+    const queryStream = this.query('mutable-store', key)
+    let found = false
+    queryStream.on('data', (result) => {
+      if (result.value === null) return
+      const { key, value, sig, seq: _seq, salt } = result.value
+      if (_seq === seq && verify(sig, value, key)) {
+        found = true
+        cb(null, value, { key, sig, seq, salt })
+        queryStream.destroy()
+      }
+    })
+
+    finished(queryStream, (err) => {
+      if (err) {
+        cb(err)
+        return
+      }
+      if (found === false) {
+        cb(Error('Not Found'))
+      }
+    })
+  },
+  put (value, opts, cb) {
+    if (!Buffer.isBuffer(value)) throw Error('Value must be a buffer')
+    if (value.length > PUT_VALUE_MAX_SIZE) { throw Error(`Value size must be <= ${PUT_VALUE_MAX_SIZE}`) }
+    if (typeof opts !== 'object') throw Error('Options are required')
+    if (typeof cb !== 'function') throw Error('Callback is required')
+    if (value.length > PUT_VALUE_MAX_SIZE) { throw Error(`Value size must be <= ${PUT_VALUE_MAX_SIZE}`) }
+    const { seq, salt, keypair } = opts
+    if (!keypair) throw Error('keypair is required')
+    const { sk, pk } = keypair
+    if (!Buffer.isBuffer(sk)) throw Error('keypair.sk (secret key buffer) is required')
+    if (!Buffer.isBuffer(pk)) throw Error('keypair.pk (public key buffer) is required')
+    if (typeof seq !== 'number') throw Error('seq is a required option')
+    if (salt) {
+      if (!Buffer.isBuffer(salt)) throw Error('salt must be a buffer')
+      if (salt.length >= 16 && salt.length <= 64) { throw Error('salt length must be between 16 and 64 bytes (inclusive)') }
+    }
+    const sig = Buffer.alloc(signSize)
+    sign(sig, value, sk)
+    const key = pk
+    const info = { key, value, sig, seq }
+
+    this.update('mutable-store', key, info, (err) => {
```

Comment:
nit: instead of these methods being implicitly bound to this === dhtNode make this a class that accepts a dht instance in the constructor. easier to read (if you ask me) and removes the need for the bind in index.js